rpcs3/Input/dualsense_pad_handler.cpp

```diff
@@ -368,6 +533,44 @@ PadHandlerBase::connection dualsense_pad_handler::update_connection(const std::s
 	return connection::connected;
 }
 
+void dualsense_pad_handler::get_extended_info(const std::shared_ptr<PadDevice>& device, const std::shared_ptr<Pad>& pad)
+{
+	auto dualsense_device = std::static_pointer_cast<DualSenseDevice>(device);
```

Comment:
std::static_pointer_cast makes a copy of shared pointer. You do **not** want to make redundent copies of shared ptrs because it's very bad for performance. Just use static_cast and .get()

Reply:
maybe in another pr

Reply:
why?

Reply:
Because it's done like this everywhere. so I'd rather "fix" it all and not just here

Reply:
It doesn't seem like perf-sensitive code to me